File: common/logging.py
import structlog
import logging
import sys
from pathlib import Path
from datetime import datetime


def setup_logging(log_dir: str = "./logs"):
    log_path = Path(log_dir)
    log_path.mkdir(parents=True, exist_ok=True)

    log_file = log_path / f"scraper-{datetime.now().strftime('%Y-%m-%d')}.log"

    structlog.configure(
        processors=[
            structlog.stdlib.add_log_level,
            structlog.stdlib.add_logger_name,
            structlog.processors.TimeStamper(fmt="iso"),
            structlog.processors.StackInfoRenderer(),
            structlog.processors.format_exc_info,
            structlog.processors.JSONRenderer()
        ],
        context_class=dict,
        logger_factory=structlog.stdlib.LoggerFactory(),
        cache_logger_on_first_use=True,
    )

    logging.basicConfig(
        format="%(message)s",
        stream=sys.stdout,
        level=logging.INFO,
    )

    file_handler = logging.FileHandler(log_file)
    file_handler.setLevel(logging.DEBUG)
    logging.getLogger().addHandler(file_handler)

    return structlog.get_logger()

File: common/src/logging.ts
import pino from 'pino';
import { mkdirSync, existsSync } from 'fs';
import { join } from 'path';

export type Logger = pino.Logger;

export function setupLogging(logDir: string = './logs'): Logger {
  const logPath = join(logDir);
  
  if (!existsSync(logPath)) {
    mkdirSync(logPath, { recursive: true });
  }

  const date = new Date().toISOString().split('T')[0];
  const logFile = join(logPath, `scraper-${date}.log`);

  const logger = pino({
    level: 'info',
    transport: {
      targets: [
        {
          target: 'pino/file',
          options: { destination: logFile },
          level: 'debug',
        },
        {
          target: 'pino-pretty',
          options: {
            colorize: true,
            translateTime: 'iso',
            ignore: 'pid,hostname',
          },
          level: 'info',
        },
      ],
    },
  });

  return logger;
}

export const logger = setupLogging();
